src/main/java/de/tum/in/www1/artemis/web/rest/hestia/ProgrammingExerciseTaskResource.java

```diff
@@ -40,6 +55,12 @@ public ProgrammingExerciseTaskResource(ProgrammingExerciseTaskRepository program
     @PreAuthorize("hasRole('TA')")
     public ResponseEntity<Set<ProgrammingExerciseTask>> getTasks(@PathVariable Long exerciseId) {
         log.debug("REST request to retrieve ProgrammingExerciseTasks for ProgrammingExercise with id : {}", exerciseId);
+        // Reload the exercise from the database as we can't trust data from the client
+        ProgrammingExercise exercise = programmingExerciseRepository.findByIdElseThrow(exerciseId);
+        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.TEACHING_ASSISTANT, exercise, null);
+
+        // updates in cases changes have been made that are not yet saved
+        programmingExerciseTaskService.updateTasksFromProblemStatement(exercise);
```

Comment:
GET should never modify anything. A developer should be able to expect that nothing gets changed on a GET call.
I really do think this update should be made when saving the exercise because that's when the problem statement gets changed.

Reply:
I agree, this is actually already implemented this way. I removed it from the method.